ui/app/components/ui/sender-to-recipient/sender-to-recipient.component.js

```diff
@@ -116,8 +142,8 @@ export default class SenderToRecipient extends PureComponent {
             <span>{ addressOnly ? `${t('to')}: ` : '' }</span>
             {
               addressOnly
-                ? checksummedRecipientAddress
-                : (recipientNickname || recipientName || this.context.t('newContract'))
+                ? (recipientNickname || recipientEns || checksummedRecipientAddress)
```

Comment:
Interesting; what is `addressOnly` supposed to do anyway? Using a nickname seems contrary to asking for the address to be used, but maybe I'm misunderstanding.

Reply:
`addressOnly` is the flag passed to the component when this is rendered in the tx details

Reply:
Right, but semantically what does it mean? "address only" implies to me that nicknames shouldn't be used, and you've just updated this to show the nickname in that case. If nicknames are desired, maybe the prop should be renamed? Or perhaps I'm misunderstanding the meaning.

Reply:
Or maybe the intended meaning was "no icons", in which case this is fine :thinking: